seq2science/rules/get_fastq.smk

```diff
@@ -176,17 +183,24 @@ rule sra2fastq_PE:
         tmpdir={config[sra_dir]}/tmp/{wildcards.sample}
         mkdir -p $tmpdir; trap "rm -rf $tmpdir" EXIT
 
-        # dump to tmp dir
-        parallel-fastq-dump -s {input}/* -O $tmpdir {config[split]} \
-        --threads {threads} --gzip >> {log} 2>&1
+        # acquire the lock
+        (
+            flock --timeout 30 200 || exit 1
+            sleep 2
+        ) 200>{layout_cachefile_lock}
+
+        # dump
+        fasterq-dump -s {input}/* -O $tmpdir --threads {threads} --split-3 >> {log} 2>&1
```

Comment:
out and tmp directories should be on the same disk, something like this?

```
# setup tmp dirs
out={config[sra_dir]}/out/{wildcards.sample}
mkdir -p $out; trap "rm -rf $out" EXIT
tmp={config[sra_dir]}/tmp/{wildcards.sample}
mkdir -p $tmp; trap "rm -rf $tmp" EXIT

...

fasterq-dump -s {input}/* -O $out -t $tmp --threads {threads} --split-3 >> {log} 2>&1

...

for f in $(ls -1q $out | grep -oP "^[^_]+" | uniq); do
```

Reply:
Yeah that would make sense